mat_dp_pipeline/data_sources/tmba.py

```diff
@@ -24,21 +24,48 @@ class TMBATargetsSource(TargetsSource):
 
     def __init__(
         self,
-        target_csv: Path,
-        targets_parameters: list[str],
+        targets: pd.DataFrame,
+        parameters: list[str],
         country_source: type[SourceWithCountries],
     ):
-        if not targets_parameters:
+        if not parameters:
             raise ValueError("You must specify parameters.")
 
-        self._targets_csv = target_csv
-        self._targets_parameters = targets_parameters
+        self._targets = targets
+        self._targets_parameters = parameters
         self._country_to_path = country_source.country_to_path(
             identifier=Identifier.alpha_2
         )
 
+    @classmethod
+    def from_excel(
+        cls,
+        spreadsheet: str | Path,
+        parameters: list[str],
+        country_source: type[SourceWithCountries],
+        sheet_name: str = "DATA_TIAM",
+        engine: Literal["xlrd", "openpyxl", "odf", "pyxlsb"] | None = None,
```

Comment:
Oh, this hasn't be changed? Unless you haven't pushed.

Reply:
It says outdated?